crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -146,6 +147,56 @@ impl Format<PyFormatContext<'_>> for FormatDanglingComments<'_> {
     }
 }
 
+/// Formats the content of the passed comment.
+///
+/// * Adds a whitespace between `#` and the comment text except if the first character is a `#`, `:`, `'`, or `!`
+/// * Replaces non breaking whitespaces with regular whitespaces except if in front of a `types:` comment
+const fn format_comment(comment: &SourceComment) -> FormatComment {
+    FormatComment { comment }
+}
+
+struct FormatComment<'a> {
+    comment: &'a SourceComment,
+}
+
+impl Format<PyFormatContext<'_>> for FormatComment<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext<'_>>) -> FormatResult<()> {
+        let slice = self.comment.slice();
+        let comment_text = slice.text(SourceCode::new(f.context().contents()));
+
+        let (content, mut start_offset) = comment_text
+            .strip_prefix('#')
+            .map_or((comment_text, TextSize::new(0)), |rest| {
+                (rest, '#'.text_len())
+            });
+
+        write!(f, [source_position(slice.start()), text("#")])?;
+
+        // Starts with a non breaking space
+        if content.starts_with('\u{A0}') && !content.trim_start().starts_with("type:") {
+            // Replace non-breaking space with a space (if not followed by a normal space)
+            start_offset += '\u{A0}'.text_len();
```

Comment:
feels like an odd rule to keep non-acsii spaces

Reply:
I took this 1:1 from black.

Reply:
yeah i meant this as a comment on black's behaviour